easypost/user.py

```diff
@@ -1,60 +1,54 @@
     @classmethod
-    def create(cls, api_key=None, **params):
-        requestor = Requestor(api_key)
+    def create(cls, api_key: str = None, **params) -> "User":
+        requestor = Requestor(local_api_key=api_key)
         url = cls.class_url()
         wrapped_params = {cls.snakecase_name(): params}
-        response, api_key = requestor.request("post", url, wrapped_params, False)
-        return util.convert_to_easypost_object(response, api_key)
+        response, api_key = requestor.request(method="post", url=url, params=wrapped_params, api_key_required=False)
+        return util.convert_to_easypost_object(response=response, api_key=api_key)
 
     @classmethod
-    def retrieve(cls, easypost_id="", api_key=None, **params):
-        try:
-            easypost_id = easypost_id["id"]
-        except (KeyError, TypeError):
-            pass
-
+    def retrieve(cls, easypost_id: str = "", api_key: str = None, **params) -> "User":
         if easypost_id == "":
-            requestor = Requestor(api_key)
-            response, api_key = requestor.request("get", cls.class_url())
-            return util.convert_to_easypost_object(response, api_key)
+            requestor = Requestor(local_api_key=api_key)
+            response, api_key = requestor.request(method="get", url=cls.class_url())
+            return util.convert_to_easypost_object(response=response, api_key=api_key)
         else:
-            instance = cls(easypost_id, api_key, **params)
+            instance = cls(easypost_id=easypost_id, api_key=api_key, **params)
             instance.refresh()
             return instance
 
     @classmethod
-    def retrieve_me(cls, api_key=None, **params):
-        requestor = Requestor(api_key)
-        response, api_key = requestor.request("get", cls.class_url())
-        return util.convert_to_easypost_object(response, api_key)
+    def retrieve_me(cls, api_key: str = None, **params) -> "User":
+        requestor = Requestor(local_api_key=api_key)
+        response, api_key = requestor.request(method="get", url=cls.class_url(), params=params)
+        return util.convert_to_easypost_object(response=response, api_key=api_key)
 
     @classmethod
-    def all_api_keys(cls, api_key=None):
-        requestor = Requestor(api_key)
-        url = "/api_keys"
-        response, api_key = requestor.request("get", url)
-        return util.convert_to_easypost_object(response, api_key)
+    def all_api_keys(cls, api_key: str = None) -> "User":
+        requestor = Requestor(local_api_key=api_key)
+        response, api_key = requestor.request(method="get", url="/api_keys")
+        return util.convert_to_easypost_object(response=response, api_key=api_key)
 
-    def api_keys(self):
+    def api_keys(self) -> Union[Union[Callable[[], List], List[Any]], Any]:
         api_keys = self.all_api_keys()
 
         if api_keys.id == self.id:
-            my_api_keys = api_keys.keys
+            return api_keys.keys
```

Comment:
Are we returning 3 times here because of `mypy` being weird? We typically try to avoid returning multiple times and return once at the end.

Reply:
Yes